auth/src/main/java/io/camunda/zeebe/auth/impl/Authorization.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.auth.impl;
+
+public class Authorization {
+
+  public static JwtAuthorizationEncoder encodeWithJwt() {
+    return new JwtAuthorizationEncoder();
+  }
```

Comment:
🙃 ❓  Why does this return the encoder, instead of the actual encoded string?

Every time we use it we now need to do:
```java
final AuthorizationEncoder encoder = Authorization.encodeWithJwt();
final String jwtToken = encoder.getEncodedString();
```

Wouldn't it be easier to just do?
```java
final String jwtToken = Authorization.encodeWithJwt();
```

Reply:
I don't think we will stray away from the defaults defined in the encoder, right? On the other hand it may be nice to keep it aligned with the decoder.

Reply:
I guess the method name is misleading. My idea was to provide a builder where we can provide the claims that we want to encode.

Currently, we only need to use the `withClaim` builder method to pass along the authorized tenants list. But in the future, we may decide to sign our JWT tokens, and we would need to pass a signing algorithm instance (ex. RSA).

Do you think the name should be changed from `encodeWithJwt()` to something else, like `buildJwtEncoder()`?

Reply:
WDYT of:

```java
Authorization.encoder().withClaim(...).encode()
```

Reply:
Or `Authorization.jwtEncoder()` in case we support other encoders in the future.

Reply:
@remcowesterhoud do you think we should align the `Authorization#decodeWithJwt()` method to be `Authorization#jwtDecorder()`?

Reply:
Yes, that's more descriptive of what it does. It doesn't decode anything. It returns an object to decode a JWT with.